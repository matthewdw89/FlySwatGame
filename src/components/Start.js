import React, { Component } from 'react';
import StartingMusic from './StartingMusic'

export default class Start extends Component {
  render() {
    return (
      <div className="start__window">
        {/* <StartingMusic></StartingMusic> */}
        <div className="start__box">
            <h1>SWAT</h1>
            <button title="The aim of the game is remove all the pesky bugs within a set timer" onClick={this.props.gameStart}>START</button>
        </div>
      </div>
    )
  }
}
