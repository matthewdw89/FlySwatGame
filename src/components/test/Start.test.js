import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Start from '../Start';

configure({ adapter: new Adapter() });

describe('<Start>', () => {

  it('should render correctly', () => {
    const component = shallow(<Start />);
    expect(component).toMatchSnapshot();
  });

  it('should register a click on the button', () => {
    let clickFn = jest.fn();
    const component = shallow(<Start gameStart={clickFn}/>);
    component.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });

});