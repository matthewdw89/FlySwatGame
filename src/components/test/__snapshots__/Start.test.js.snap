// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Start> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Start />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="start__box"
      >
        <h1>
          SWAT
        </h1>
        <button
          title="The aim of the game is remove all the pesky bugs within a set timer"
        >
          START
        </button>
      </div>,
      "className": "start__window",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SWAT
          </h1>,
          <button
            title="The aim of the game is remove all the pesky bugs within a set timer"
          >
            START
          </button>,
        ],
        "className": "start__box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAT",
          },
          "ref": null,
          "rendered": "SWAT",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "START",
            "onClick": undefined,
            "title": "The aim of the game is remove all the pesky bugs within a set timer",
          },
          "ref": null,
          "rendered": "START",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="start__box"
        >
          <h1>
            SWAT
          </h1>
          <button
            title="The aim of the game is remove all the pesky bugs within a set timer"
          >
            START
          </button>
        </div>,
        "className": "start__window",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              SWAT
            </h1>,
            <button
              title="The aim of the game is remove all the pesky bugs within a set timer"
            >
              START
            </button>,
          ],
          "className": "start__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWAT",
            },
            "ref": null,
            "rendered": "SWAT",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "START",
              "onClick": undefined,
              "title": "The aim of the game is remove all the pesky bugs within a set timer",
            },
            "ref": null,
            "rendered": "START",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
